core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3213,7 +3213,14 @@ public void activeSyncMetadata(AlluxioURI path, Collection<AlluxioURI> changedFi
 
     Map<AlluxioURI, UfsStatus> statusCache;
     try (RpcContext rpcContext = createRpcContext()) {
-      statusCache = populateStatusCache(path, DescendantType.ALL);
+      if (changedFiles == null) {
+        // full sync
+        statusCache = populateStatusCache(path, DescendantType.ALL);
+      } else {
+        // incremental sync
+        statusCache = populateStatusCache(PathUtils.findLowestCommonAncestor(changedFiles),
```

Comment:
does `populateStatusCache` handle `null` paths?

Reply:
nope, findLCA is never null though for non-empty change list (the incremental case)